src/responder/server-nsq.ts

```diff
@@ -115,14 +125,34 @@ export class ServerNsq extends Server implements CustomTransportStrategy {
             topic,
             channel,
           });
+          let source;
           try {
-            await handler(packet.data, nsqCtx);
-            msg.finish();
+            source = await handler(packet.data, nsqCtx);
+            if (!isObservable(source)) {
+              return msg.finish();
+            }
           } catch (err) {
             this.logger.error(
-              `consumer reader failed to process message with error: ${err}, topic: ${topic}, channel: ${channel}`,
+              `consumer reader failed to process message with error: ${err.message}, topic: ${topic}, channel: ${channel}`,
             );
-            msg.requeue(this.options.requeueDelay || 90 * 1000);
+            msg.requeue(consumerOptions.requeueDelay || DEFAULT_REQUEUE_DELAY);
```

Comment:
Guess should `return` here? Otherwise `source` might be empty when being called with `.pipe` method later.

Reply:
fixing